File: code/classes/subclass.ex
# Subclass in Elixir (no direct equivalent, use structs and protocol delegation)
defmodule Person do
  defstruct name: nil, age: nil

  def new(name, age), do: %Person{name: name, age: age}

  def greet(%Person{name: name, age: age}) do
    "Hello, I'm #{name} and I'm #{age} years old"
  end
end

defmodule Student do
  defstruct name: nil, age: nil, school: nil

  def new(name, age, school), do: %Student{name: name, age: age, school: school}

  def greet(%Student{name: name, age: age, school: school}) do
    person = Person.new(name, age)
    Person.greet(person) &lt;&gt; " I study at #{school}"
  end
end

student = Student.new("Alice", 20, "MIT")
IO.puts Student.greet(student)

File: code/classes/subclass.erl
% Subclass in Erlang (no direct equivalent, use records and module delegation)
-module(subclass).
-export([new_person/2, new_student/3, greet/1, start/0]).

-record(person, {name, age}).
-record(student, {person, school}).

new_person(Name, Age) -&gt;
    #person{name=Name, age=Age}.

new_student(Name, Age, School) -&gt;
    #student{person=new_person(Name, Age), school=School}.

greet(#person{name=Name, age=Age}) -&gt;
    io_lib:format("Hello, I'm ~s and I'm ~p years old", [Name, Age]);
greet(#student{person=P, school=School}) -&gt;
    [greet(P), io_lib:format(" I study at ~s", [School])].

start() -&gt;
    Student = new_student("Alice", 20, "MIT"),
    io:format("~s~n", [greet(Student)]).
